rest/client/request.go

```diff
@@ -201,9 +199,8 @@ func (c *communicatorImpl) retryRequest(ctx context.Context, info requestInfo, d
 			} else if resp.StatusCode == http.StatusConflict {
 				return nil, HTTPConflictError
 			} else if resp.StatusCode >= 400 && resp.StatusCode < 500 {
-				apiErr := &rest.APIError{}
-				if err := util.ReadJSONInto(resp.Body, err); err != nil {
-					return nil, errors.Wrapf(apiErr, "server returned %d", resp.StatusCode)
+				if bytes, _ := ioutil.ReadAll(resp.Body); len(bytes) > 0 {
```

Comment:
this is probably fine, but I think that we need to avoid unprotected/limited reading from request bodies. so something like util.NewRequestReader() should do it. 

Having said that, ioutil.ReadAll doesn't close the body (something should).